### PR TITLE
fix: protocol should not be part of the volume

### PR DESCRIPTION
**Pull request by tiagolobocastro**
The share protocol should not be part of the volume as it's decided on publish.
In the future we may even do things such as publish the volume with a protocol
that is suited for the node where the application runs.

So, instead make the protocol part of the volume target (ie the nexus).

Also, make the state a non optional component of the volume so that we do not
need to "fake" it on multiple layers, instead the core agent will set it.

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
bors try

File: common/src/types/v0/message_bus/volume.rs [hunk 1]
```diff
@@ -39,18 +36,18 @@ impl Volume {
...
     pub fn status(&self) -> Option<VolumeStatus> {
-        self.state.as_ref().map(|s| s.status.clone())
+        Some(self.state.status.clone())
     }
 }
 
 impl From<Volume> for models::Volume {
     fn from(volume: Volume) -> Self {
-        models::Volume::new_all(volume.spec(), volume.state().into_opt())
+        models::Volume::new_all(volume.spec(), volume.state)
```
Comment (mtzaurus) [inline common/src/types/v0/message_bus/volume.rs:50]: Shouldn't we have the same behaviour here: either move both members out of target volume or use getters ?

Comment (tiagolobocastro) [reply common/src/types/v0/message_bus/volume.rs:50]: Yes that should be state()

Comment (tiagolobocastro) [reply common/src/types/v0/message_bus/volume.rs]: Done

**Review by mtzaurus — APPROVED** [review]

**Review by paulyoong — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**